modules/doi/includes/utilities.inc

```diff
@@ -864,3 +862,51 @@ function islandora_doi_get_mods($id) {
 
   return islandora_doi_translate($source_string, $content_type);
 }
+
+/**
+ * Filter HTML tags from imported XML metadata.
+ *
+ * @param DOMNode $node
+ *   The node containing metadata that should be converted into a string,
+ *   possibly containing markup.
+ *
+ * @return string 
+ *   The node's inner content with markup escaped or stripped.
+ */
+function islandora_doi_filter_html($node, $action = NULL) {
+  if (!$node) {
+    return "";
+  }
+  else {
+    $text  = '';
+    foreach ($node->childNodes as $child) {
+      $text .= $child->ownerDocument->saveXML($child);
+    }
+
+    $text = preg_replace('/\n\s+/', ' ', $text);
+
+    if (!$action) {
+      $action = variable_get('islandora_doi_handle_imported_metadata', 'strip');
+    }
+
+    switch ($action) {
+      case 'strip':
+        $text = strip_tags($text);
+        break;
+
+      case 'encode':
+        $text = htmlspecialchars($text);
```

Comment:
Not sure at all, but are the default flags and default encoding always what we want? I am sure this will strongly depend on user setup, but perhaps we want to think a moment about the implications? See [`htmlspecialchars()`](https://secure.php.net/htmlspecialchars) for reference.